docs/data/rpc/api-reference/methods/getLedgerEntries.mdx

```diff
@@ -48,291 +145,191 @@ print(
+- When you deploy a contract, first the code is "installed" (i.e. uploaded onto the blockchain), creating a `LedgerEntry` with the WASM byte-code that can be uniquely identified by its hash (that is, the hash of the uploaded code itself).
+- Then, when a contract _instance_ is "deployed," we create a `LedgerEntry` with a reference to that code's hash. This means many contracts can point to the same WASM code.
 
-```json
-{
-  "jsonrpc": "2.0",
-  "id": 8675309,
-  "method": "getLedgerEntries",
-  "params": {
-    "keys": ["AAAAAAAAAACp3BPIqFxM9XSnW6aHvavD3GWlJGfuylOt5tZL6CQtdQ=="]
-  }
-}
-```
-
-And the response we get contains the `LedgerEntryData` with the current information about this account.
-
-```json
-{
-  "jsonrpc": "2.0",
-  "id": 8675309,
-  "result": {
-    "entries": [
-      {
-        "key": "AAAAAAAAAACp3BPIqFxM9XSnW6aHvavD3GWlJGfuylOt5tZL6CQtdQ==",
-        "xdr": "AAAAAAAAAACp3BPIqFxM9XSnW6aHvavD3GWlJGfuylOt5tZL6CQtdQAAABdIdugAAAWpygAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAA",
-        "lastModifiedLedgerSeq": "164303"
-      }
-    ],
-    "latestLedger": 246819
-  }
-}
-```
-
-We can then parse this result as an `xdr.LedgerEntryData` type.
-
-```js
-const parsed = xdr.LedgerEntryData.fromXDR(
-  "AAAAAAAAAACp3BPIqFxM9XSnW6aHvavD3GWlJGfuylOt5tZL6CQtdQAAABdIdugAAAWpygAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAA",
-  "base64",
-);
-console.log(parsed);
-```
-
-### Requesting a Contract's Wasm Code
-
-This can be a bit tricky to wrap your head around, but the conventions do make sense once you let it sink in.
-
-In the previous examples, the `COUNTER` _key_ was used as a `LedgerKey` while the incremented _value_ was stored in a **`LedgerEntry`**. "Ledger Entry" is the relevant term to keep in mind during this discussion. That `LedgerEntry` was stored on the Stellar ledger, and was associated with a corresponding `LedgerKey`. `LedgerKey: LedgerEntry` works the same way you would think of almost any `key: value` storage system.
-
-#### How Soroban Contract Deployment Works
-
-When you deploy a contract, first the code is "installed" (i.e. it is uploaded onto the blockchain). This creates a `LedgerEntry` containing the Wasm byte-code, which is uniquely identified by its hash (that is, the hash of the uploaded code itself). Then, when the contract is "deployed," we create a `LedgerEntry` with a reference to that code's hash. So fetching the contract code is a two-step process:
+Thus, fetching the contract code is a two-step process:
 
 1. First, we look up the contract itself, to see which code hash it is referencing.
-2. Then, we can look up the raw Wasm byte-code using that hash.
+2. Then, we can look up the raw WASM byte-code using that hash.
 
-#### Request the `LedgerKey` for the Contract Code
+#### 1. Find the ledger key for the contract instance
 
-##### Python
+<CodeExample>
 
 ```python
```

Comment:
Some examples are Python and JavaScript, some are Python and TypeScript, and some are unlabelled and I assume just JavaScript. Can we make them consistent, maybe all JS instead of TS, and all also have Python?